sound/soc/sof/intel/cnl.c

```diff
@@ -173,10 +173,6 @@ static int cnl_ipc_send_msg(struct snd_sof_dev *sdev,
 {
 	struct sof_intel_hda_dev *hdev = sdev->pdata->hw_pdata;
 	struct sof_ipc_cmd_hdr *hdr;
-	const struct sof_dsp_power_state target_state = {
-		.state = SOF_DSP_PM_D0,
-		.substate = SOF_HDA_DSP_PM_D0I0,
-	};
 	u32 dr = 0;
```

Comment:
could you update the commit message and clarify in which circumstances we would suspend with a mutex lock held?

Reply:
@plbossart updated the commit message now. Its not that we suspend with the mutex help but rather that we try to send another IPC while the lock is already held

Reply:
The wording 

But, this would result in a deadlock because the
snd_sof_dsp_send_msg() op is invoked while the IPC
tx_mutex and ipc_lock are both already held
and the snd_sof_dsp_set_power_save() op will
initiate a new IPC and try to acquire the tx_mutex.

is not great.

Maybe use a sequence such as 

mutex_lock
snd_sof_send_message
    snd_sof_dsp_set_power_save
        mutex_lock << DEADLOCK.

Reply:
nice, thanks! updated now